lib/onelogin/ruby-saml/response.rb

```diff
@@ -184,8 +195,8 @@ def validate_response_state(soft = true)
       end
 
       def xpath_first_from_signed_assertion(subelt=nil)
-        node = REXML::XPath.first(document, "/p:Response/a:Assertion[@ID='#{document.signed_element_id}']#{subelt}", { "p" => PROTOCOL, "a" => ASSERTION })
-        node ||= REXML::XPath.first(document, "/p:Response[@ID='#{document.signed_element_id}']/a:Assertion#{subelt}", { "p" => PROTOCOL, "a" => ASSERTION })
+        node = REXML::XPath.first(assertion_document, "/samlp:Response/saml:Assertion[@ID='#{assertion_document.signed_element_id}']#{subelt}", { "samlp" => PROTOCOL, "saml" => ASSERTION, 'ds' => DSIG })
```

Comment:
this change does not incorporate the security fix from 0.8.2

https://github.com/onelogin/ruby-saml/blob/master/lib/onelogin/ruby-saml/response.rb#L186-L200